flash/image/classification/adapters.py

```diff
@@ -268,7 +268,11 @@ def _convert_dataset(
             devices = 1
             if isinstance(trainer.training_type_plugin, DataParallelPlugin):
                 # when using DP, we need to sample n tasks, so it can splitted across multiple devices.
-                devices = trainer.accelerator_connector.devices
+                if hasattr(trainer, "accelerator_connector"):
+                    accelerator_connector = trainer.accelerator_connector
+                else:
+                    accelerator_connector = trainer._accelerator_connector
```

Comment:
```suggestion
                if hasattr(trainer, "_accelerator_connector"):
                    accelerator_connector = trainer._accelerator_connector
                else:
                    accelerator_connector = trainer.accelerator_connector
```

Try using the new API first and just fall back to legacy if necessary :)

Reply:
Done :smiley: